packages/flutter/lib/src/widgets/decorated_sliver.dart

```diff
@@ -25,15 +25,28 @@ import 'image.dart';
 ///
 /// Commonly used with [BoxDecoration].
 ///
-/// The [child] is not clipped. To clip a child to the shape of a particular
-/// [ShapeDecoration], consider using a [ClipPath] widget.
 ///
 /// {@tool dartpad}
 /// This sample shows a radial gradient that draws a moon on a night sky:
 ///
 /// ** See code in examples/api/lib/widgets/sliver/decorated_sliver.0.dart **
 /// {@end-tool}
 ///
+/// {@tool dartpad}
+/// This is a height-resizable window simulating a browser window with [CustomScrollView].
+/// [Slider] adjusts the height of window.
+/// [Switch] toggles the [clipBehavior] of [CustomScrollView].
+///
+/// Demonstrates the behavior of [DecoratedSliver] in relation to parent clipping.
+///
+/// ** See code in examples/api/lib/widgets/sliver/decorated_sliver.1.dart **
+/// {@end-tool}
+///
+/// [clipBehavior] is set by parent [Widget].
+///
+/// To clip a child to the shape of a
+/// particular [ShapeDecoration], consider using a [ClipPath] widget.
```

Comment:
```suggestion
/// This widget does not apply any additional clipping to its [child].
/// To clip a child based on the [Decoration]'s shape, consider using
/// a [ClipPath] widget.
```

([relevant style guideline](https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#use-correct-grammar))